indico/modules/receipts/client/js/printing/PrintReceiptsModal.jsx

```diff
@@ -0,0 +1,303 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2023 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import allTemplatesURL from 'indico-url:receipts.all_templates';
+import exportReceiptsURL from 'indico-url:receipts.receipts_export';
+import previewReceiptsURL from 'indico-url:receipts.receipts_preview';
+
+import PropTypes from 'prop-types';
+import React, {useState} from 'react';
+import {Field, FormSpy} from 'react-final-form';
+import {Button, Dropdown, Form, Grid, Header, Icon, Message, Segment} from 'semantic-ui-react';
+
+import {
+  formatters,
+  FinalCheckbox,
+  FinalDropdown,
+  FinalField,
+  FinalInput,
+  FinalSubmitButton,
+} from 'indico/react/forms';
+import {FinalModalForm} from 'indico/react/forms/final-form';
+import {useIndicoAxios} from 'indico/react/hooks';
+import {Param, Plural, PluralTranslate, Singular, Translate} from 'indico/react/i18n';
+import {handleAxiosError, indicoAxios} from 'indico/utils/axios';
+import {downloadBlob} from 'indico/utils/browser';
+import {snakifyKeys} from 'indico/utils/case';
+
+import Previewer from '../templates/Previewer.jsx';
+
+import {printReceipt} from './print.jsx';
+import TemplateParameterEditor from './TemplateParameterEditor.jsx';
+
+const makeSubmitLabel = ({publish, notify_users: notifyUsers}) => {
+  if (publish) {
+    return notifyUsers
+      ? Translate.string('Publish and send to registrant(s)')
+      : Translate.string('Save and publish to registration(s)');
+  }
+  return notifyUsers
+    ? Translate.string('Save and send to registrant(s)')
+    : Translate.string('Save to registration(s)');
+};
+
+const getDefaultValue = f =>
+  f.type === 'dropdown' ? f.attributes.options[f.attributes.value] : f.attributes.value;
+
+const downloadOptions = [
+  {
+    key: 'pdf',
+    value: 'pdf',
+    text: Translate.string('as a single PDF document'),
+  },
+  {
+    key: 'zip',
+    value: 'zip',
+    text: Translate.string('as a ZIP archive'),
+  },
+];
+
+const archiveFilename = {
+  pdf: 'documents.pdf',
+  zip: 'documents.zip',
```

Comment:
Would it make sense for these to be translatable?

Reply:
We don't translate filenames anywhere else... I already find it very ugly when people share screenshots and it's a "capture de pantalla.png" or something like that.